### ISSUE TITLE
pihole no longer blocking ads

### ISSUE THREAD

**Issue by gottabeleGIT** [pi-hole/pi-hole#3390]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [] The issue I am reporting can be *replicated*. - **Maybe...? (unsure)**
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`1`

---
**Expected behaviour:**

`Ads and blacklisted sites unable to load on devices using the raspberry pi with pihole as a DNS server.`

**Actual behaviour:**

`Ads and blacklisted sites are no longer blocked.`

**Steps to reproduce:**

`It started about a day after upgrading fom piHole 4.3 to 5.0. I haven't recently changed anything on the raspberry pi at all besides the upgrade. Apache2 is also running on the pi at port 80, so I mapped lighttpd to run on port 8080.`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`https://tricorder.pi-hole.net/0o1tylpxxw`

**Troubleshooting undertaken, and/or other relevant information:**

`{Steps of what you have done to fix this}`

> * Restart raspberry pi
> * Change DNS settings on router away from and back to the Pi
> * Installed `php-sqlite3`
> * Re-installed piHole (not clean install)
> * Restart router
> * Disabled and re-enabled blacklists 
> * Adding devices to groups
> * Updated gravity lists
> * Changed upstream DNS on pihole (from Cloudflare to Quad9)

**Comment by MarsRaccoon** [comment]
Have you made sure your devices are actually using pi-hole? and not another DNS server.

**Comment by gottabeleGIT** [comment]
Yes

**Comment by dschaper** [comment]
Check them again to make sure, the debug log shows no clients are requesting DNS from it.

```plain
   [2020-05-15 17:24:16.708 680] Database successfully initialized
   [2020-05-15 17:24:16.708 680]  -> Total DNS queries: 0
   [2020-05-15 17:24:16.708 680]  -> Cached DNS queries: 0
   [2020-05-15 17:24:16.708 680]  -> Forwarded DNS queries: 0
   [2020-05-15 17:24:16.709 680]  -> Blocked DNS queries: 0
   [2020-05-15 17:24:16.709 680]  -> Unknown DNS queries: 0
   [2020-05-15 17:24:16.709 680]  -> Unique domains: 0
   [2020-05-15 17:24:16.709 680]  -> Unique clients: 0
   [2020-05-15 17:24:16.709 680]  -> Known forward destinations: 0
```

```plain
*** [ DIAGNOSING ]: Pi-hole log
-rw-r--r-- 1 pihole pihole 782 May 15 17:24 /var/log/pihole.log
   -----head of pihole.log------
   May 15 17:23:48 dnsmasq[683]: exiting on receipt of SIGTERM
   May 15 17:24:16 dnsmasq[682]: started, version pi-hole-2.81 cachesize 10000
   May 15 17:24:16 dnsmasq[682]: compile time options: IPv6 GNU-getopt no-DBus no-UBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset auth DNSSEC loop-detect inotify dumpfile
   May 15 17:24:16 dnsmasq[682]: using only locally-known addresses for domain use-application-dns.net
   May 15 17:24:16 dnsmasq[682]: using nameserver 149.112.112.10#53
   May 15 17:24:16 dnsmasq[682]: using nameserver 9.9.9.10#53
   May 15 17:24:16 dnsmasq[682]: read /etc/hosts - 5 addresses
   May 15 17:24:16 dnsmasq[682]: failed to load names from /etc/pihole/custom.list: No such file or directory
   May 15 17:24:16 dnsmasq[682]: read /etc/pihole/local.list - 4 addresses
```

**Comment by gottabeleGIT** [comment]
I have a device set to the pi for DNS right now, whatsmydnsserver.com shows up as Quad9, which is what the pihole upstream DNS is set to, whereas my router is currently on AdGuard. Ads are still showing up on the device connected to pihole...

**Comment by dschaper** [comment]
On one of your clients please run `dig pi.hole` and provide the response.

**Comment by gottabeleGIT** [comment]
![Screen Shot 2020-05-15 at 3 40 11 PM](https://user-images.githubusercontent.com/21181118/82094358-6dfd8180-96ec-11ea-86d8-214f2ab3eaa5.jpg)

**Comment by dschaper** [comment]
Good, and now try `dig flurry.com` (or any domain that should be blocked.)

**Comment by gottabeleGIT** [comment]
Some domains that I know should be blocked:

![Screen Shot 2020-05-15 at 4 19 49 PM](https://user-images.githubusercontent.com/21181118/82097261-13ffba80-96f2-11ea-8207-030d75daad59.jpg)
![Screen Shot 2020-05-15 at 4 20 03 PM](https://user-images.githubusercontent.com/21181118/82097318-31cd1f80-96f2-11ea-9734-7c8f4cbd4c3e.jpg)
![Screen Shot 2020-05-15 at 4 20 11 PM](https://user-images.githubusercontent.com/21181118/82097358-44475900-96f2-11ea-82ad-2b4871aafab6.jpg)

**Comment by dschaper** [comment]
It appears there is not blocking enabled on the Pi-hole. 

Does `pihole -q flurry.com` return any list or hits?

**Comment by gottabeleGIT** [comment]
![Screen Shot 2020-05-15 at 4 25 38 PM](https://user-images.githubusercontent.com/21181118/82097531-9a1c0100-96f2-11ea-9366-bb54c36f7902.jpg)

**Comment by dschaper** [comment]
Try `pihole -q -exact flurry.com` to narrow it down.

**Comment by gottabeleGIT** [comment]
![Screen Shot 2020-05-15 at 4 27 06 PM](https://user-images.githubusercontent.com/21181118/82097625-ca639f80-96f2-11ea-9de0-0bb30f7018b6.jpg)

**Comment by dschaper** [comment]
The next step would be to enable logging and tail the log to see what FTL is actually doing in the background.

**Comment by gottabeleGIT** [comment]
I enabled logging and visited a few sites, the log is blank...

**Comment by dschaper** [comment]
New debug please?

**Comment by gottabeleGIT** [comment]
https://tricorder.pi-hole.net/iovsapdp1r

**Comment by PromoFaux** [comment]
Just spotted in your debug log ` [2020-05-15 17:24:16.701 680]    PRIVACYLEVEL: Set to 4`

This is the reason nothing is blocking. Something we didn't realise until after we released 5.0, level 4 will be entirely disabled once we release 5.1

From this documentation change here https://github.com/pi-hole/docs/pull/310

>Remove privacy level 4 as blocking does not work in this mode (we will remove it for v5.1). No harm to remove the documentation already now. Disabling any processing effectively disables blocking which is counterproductive.

Drop privacy level to 3 and see if that resolves things

**Comment by gottabeleGIT** [comment]
Yes, that seems to have fixed it, thanks! :D
